examples/meander/meander_example.py

```diff
@@ -0,0 +1,157 @@
+"""
+Meander Test case
+=======================
+
+Solves the test case of a meander.
+
+We use this test case to validate the implementation of the mathematical and
+numerical methods used in Thetis to model sediment transport and morphological changes.
+Specifically this test case tests the secondary current implementation.
+
+For more details, see
+[1] Clare et al. 2020. “Hydro-morphodynamics 2D Modelling Using a Discontinuous
+    Galerkin Discretisation.” EarthArXiv. January 9. doi:10.31223/osf.io/tpqvy.
+"""
+
+from thetis import *
+
+import time
+import datetime
+
+# Note it is necessary to run meander_hydro first to get the hydrodynamics simulation
+
+
+def update_forcings_bnd(t_new):
+
+    if t_new != t_old.dat.data[:]:
+        # update boundary condtions
+        if t_new*morfac <= 6000:
+            elev_constant.assign(gradient_elev*t_new*morfac + elev_init_const)
+            flux_constant.assign((gradient_flux*t_new*morfac) - 0.02)
+        else:
+            flux_constant.assign((gradient_flux2*(t_new*morfac-6000)) - 0.053)
+            elev_constant.assign(gradient_elev2*(t_new*morfac-18000) + elev_init_const)
+        t_old.assign(t_new)
+
+
+t_old = Constant(0.0)
+
+# define mesh
+mesh2d = Mesh("meander.msh")
+x, y = SpatialCoordinate(mesh2d)
+
+# define function spaces
+V = FunctionSpace(mesh2d, 'CG', 1)
+DG_2d = FunctionSpace(mesh2d, 'DG', 1)
+vector_dg = VectorFunctionSpace(mesh2d, 'DG', 1)
+
+# define underlying bathymetry
+
+bathymetry_2d = Function(V, name='Bathymetry')
+
+gradient = Constant(0.0035)
+
+L_function = Function(V).interpolate(conditional(x > 5, pi*4*((pi/2)-acos((x-5)/(sqrt((x-5)**2+(y-2.5)**2))))/pi, pi*4*((pi/2)-acos((-x+5)/(sqrt((x-5)**2+(y-2.5)**2))))/pi))
+bathymetry_2d1 = Function(V).interpolate(conditional(y > 2.5, conditional(x < 5, (L_function*gradient) + 9.97072, -(L_function*gradient) + 9.97072), 9.97072))
+
+init = max(bathymetry_2d1.dat.data[:])
+final = min(bathymetry_2d1.dat.data[:])
+
+bathymetry_2d2 = Function(V).interpolate(conditional(x <= 5, conditional(y <= 2.5, -9.97072 + gradient*abs(y - 2.5) + init, 0), conditional(y <= 2.5, -9.97072 - gradient*abs(y - 2.5) + final, 0)))
+bathymetry_2d = Function(V).interpolate(-bathymetry_2d1 - bathymetry_2d2)
+
+# record initial bathymetry before it evolves
+initial_bathymetry_2d = Function(V).interpolate(bathymetry_2d)
+
+# choose directory to output results
+ts = time.time()
+st = datetime.datetime.fromtimestamp(ts).strftime('%Y-%m-%d %H:%M:%S')
+outputdir = 'outputs' + st
+
+# initialise velocity and elevation
+chk = DumbCheckpoint("hydrodynamics_meander/elevation", mode=FILE_READ)
```

Comment:
Maybe not needed in this PR, but would be good to find a way not to have to read the init condition from the disk, especially when running the test suite.